### ISSUE TITLE
Incorrect BTC/GH/Day?

### ISSUE THREAD

**Issue by luismcv** [aaronsace/MultiPoolMiner#482]
I'm not sure what was going on. Maybe a temporary spike on profits made the stats go awry and they take a very long time to get back to normal?

I'm using release v2.6.0.2 so this might have been fixed already in the master branch. Today I had to remove Keccak from the list of algorithms after noticing it had been mining it for 17 hours because the reported BTC/GH/Day was double what it should be. 

The Mining Pool Hub's page was showing a non-normalised BTC/GH/Day of 0.00030.

The contents of MiningPoolHub_Keccak_Profit.txt were:

```
{
    "Live":  0.00000000000030463,
    "Minute":  0.00000000000030463,
    "Minute_Fluctuation":  1.6573252120324576E-16,
    "Minute_5":  0.000000000000304724958859281,
    "Minute_5_Fluctuation":  0.0014818672495582816,
    "Minute_10":  0.000000000000305260998124119,
    "Minute_10_Fluctuation":  0.0039586985787417915,
    "Hour":  0.000000000000297464309196876,
    "Hour_Fluctuation":  0.083525192329527015,
    "Day":  0.000000000000610794375934689,
    "Day_Fluctuation":  1.4807463598154402,
    "Week":  0.000000000000387616871646063,
    "Week_Fluctuation":  0.70829290347086549,
    "Updated":  "\/Date(1510002010089)\/"
}
```

All the profit rates seem pretty much reasonable, except the daily one, that for some reason is double all the more recent ones. I guess it's the one that MultiPoolMiner reports and uses. I checked my total profits in MPM since yesterday and, indeed, they were much lower than what they usually are (a third or so), so it's not that Keccak had really been that profitable most of the last day.

I think that, at least in the meantime, I'm going to leave enabled only the few algorithms that I know are most profitable most of the time with my 2 GPUs.

Edit: I said before that I had to remove Cryptonight the other day too, but it wasn't Cryptonight (can't remember which one though). The reason I had to remove Cryptonight had nothing to do with this, it was because Nicehash's servers weren't responding and MultiPoolMiner kept retrying for hours.

**Comment by FruitFace** [comment]
I wonder if something is wrong with MPH.  When using mininpoolhubcoins pool, MPH makes the algo mining decision, while If I use miningpoolhub pool, MPM chooses what to mine.  Keccac was tops when using miningpoolhubcoins, but now, its near the bottom, when using miningpoolhub.

**Comment by fonyo** [comment]
It is the other way around, miningpoolhubcoins will grant multipoolminer the decision to choose the coin.

**Comment by FruitFace** [comment]
Ah ok, so then something might be wonky with MPM after all.

**Comment by fonyo** [comment]
It might have switched during when you switched between MPM configs. Keccak is one of those that jumps up and down quickly.

**Comment by FruitFace** [comment]
Funny thing was MPM had mined keccak for hours, and was still mining it, until I cleared it from my algo list.  I haven't added it back yet, to see if it's corrected.

**Comment by fonyo** [comment]
This is a difficult one. A good while back I was testing a case that involved running the resetprofit.bat every 5 minutes. This way you only have current and 5-minute-average. The issue with this is that you are not safeguarded from spikes in a sense, however, it would reset itself to normal after 5 minutes. When a spike occurs, the best is to delete the profit.txt and the one file affected by the spike (in this case, it is keccak) so you can keep you averages and keep MPM becoming "smarter".

**Comment by luismcv** [comment]
I don't know about FruitFace's, but in my case I wasn't switching between configs. The fact that at the very least both FruitFace and I were affected might indicate that there is (was? again, at least in my case, it was release 2.6.0.2, not the current code) something odd with the profit stats. 

Besides, I don't understand how a peak could affect the Day's stats by so much. It would have required Keccak prices to be either extremely high if it was for a short time or not so high but for a long time (which was not the case, as my MPM profits during the previous 24 hours or so were really low).

In any case, I've seen that aaronsace commited a change yesterday, changing the prices the profits for MiningPoolHub are based on from Day to Live? That would, I guess, alleviate the problem; although it might make MPM to switch algorithms too frequently after every small spike? Or maybe not (I haven't gone too deep into the scripts' logic, I've never used PowerShell and I get lost easily with it) because there seem to be some calculations (_Comparison, _MarginOfError, _Bias) that probably prevent this? A little explanation of what each one is, how they're calculated and how they affect the switch decisions would be appreciated :)

On a different note, is the current master branch stable enough? A month ago or so when I first started using MPM I tried the most current code at the time, and there were some serious issues so I just went for the latest release instead. I might give it another go again. Or better wait if a new release is coming soon?

**Comment by FruitFace** [comment]
Full disclosure, I had, a while back updated using the file commits, up to,
and including the watchdog update on Oct 15.  I have not updated since
then.

Didn't think of it until now...

On Nov 9, 2017 2:08 PM, "luismcv" <notifications@github.com> wrote:

I don't know about FruitFace's, but in my case I wasn't switching between
configs. The fact that at the very least both FruitFace and I were affected
might indicate that there is (was? again, at least in my case, it was
release 2.6.0.2, not the current code) something odd with the profit stats.

Besides, I don't understand how a peak could affect the Day's stats by so
much. It would have required Keccak prices to be either extremely high if
it was for a short time or not so high but for a long time (which was not
the case, as my MPM profits during the previous 24 hours or so were really
low).

In any case, I've seen that aaronsace commited a change yesterday, changing
the prices the profits for MiningPoolHub are based on from Day to Live?
That would, I guess, alleviate the problem; although it might make MPM to
switch algorithms too frequently after every small spike? Or maybe not (I
haven't gone too deep into the scripts' logic, I've never used PowerShell
and I get lost easily with it) because there seem to be some calculations
(_Comparison, _MarginOfError, _Bias) that probably prevent this? A little
explanation of what each one is, how they're calculated and how they affect
the switch decisions would be appreciated :)

On a different note, is the current master branch stable enough? A month
ago or so when I first started using MPM I tried the most current code at
the time, and there were some serious issues so I just went for the latest
release instead. I might give it another go again. Or better wait if a new
release is coming soon?

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/aaronsace/MultiPoolMiner/issues/482#issuecomment-343291866>,
or mute the thread
